api-http/proposals/http-plugin.md

```diff
@@ -0,0 +1,185 @@
+
+### chain_ro
+  - v1/chain/get_activated_protocol_features
+  - v1/chain/get_block_info
+  - v1/chain/get_block_header_state
+  - v1/chain/get_account
+  - v1/chain/get_code
+  - v1/chain/get_code_hash
+  - v1/chain/get_consensus_parameters
+  - v1/chain/get_abi
+  - v1/chain/get_raw_code_and_abi
+  - v1/chain/get_raw_abi
+  - v1/chain/get_table_rows
+  - v1/chain/get_table_by_scope
+  - v1/chain/get_currency_balance
+  - v1/chain/get_currency_stats
+  - v1/chain/get_producers
+  - v1/chain/get_producer_schedule
+  - v1/chain/get_scheduled_transactions
+  - v1/chain/get_required_keys
+  - v1/chain/get_transaction_id
+  - v1/chain/send_read_only_transaction
+  - v1/chain/get_accounts_by_authorizers
+  - v1/chain/get_raw_block
+  - v1/chain/get_block
+  - v1/chain/get_block_header
+  - v1/chain/get_transaction_status
+
+### chain_rw
+  - v1/chain/compute_transaction
+  - v1/chain/push_transaction
+  - v1/chain/push_transactions
+  - v1/chain/send_transaction
+  - v1/chain/send_transaction2
+
+### db_size
+  - v1/db_size/get
+### net_ro
+  - v1/net/status
+  - v1/net/connections
+
+### net_rw
+  - v1/net/connect
+  - v1/net/disconnect
+
+### producer_ro
+  - v1/producer/paused
+  - v1/producer/get_runtime_options
+  - v1/producer/get_greylist
+  - v1/producer/get_whitelist_blacklist
+  - v1/producer/get_scheduled_protocol_feature_activations
+  - v1/producer/get_supported_protocol_features
+  - v1/producer/get_account_ram_corrections
+  - v1/producer/get_unapplied_transactions
+
+### producer_rw
+  - v1/producer/pause
+  - v1/producer/resume
+  - v1/producer/update_runtime_options
+  - v1/producer/add_greylist_accounts
+  - v1/producer/remove_greylist_accounts
+  - v1/producer/set_whitelist_blacklist
+  - v1/producer/schedule_protocol_feature_activations
+  - v1/producer/get_integrity_hash
+
+### snapshot
+  - v1/producer/get_snapshot_requests
+  - v1/producer/create_snapshot
+  - v1/producer/schedule_snapshot
+  - v1/producer/unschedule_snapshot
+
+### trace_api
+  - v1/trace_api/get_block
+  - v1/trace_api/get_transaction_trace
+
+### prometheus
+  - v1/prometheus/metrics
+
+### node
+  - v1/node/get_supported_apis
+  - v1/chain/get_info
+
+## User configuration knob
+A single `http-category-address` can be used to configure all addresses in command line and ini file. The option can be used multiple times as needed.
+
+```config.ini
+http-server-address   = http-category-address
+http-category-address = chain_ro,127.0.0.1:8081
+http-category-address = chain_ro,localhost:9090
+http-category-address = chain_rw,[::1]:8082
+http-category-address = net_ro,[::1]:8081
+http-category-address = producer_ro,/tmp/absolute_unix_path.sock
+http-category-address = wallet,./relative/unix_path.sock
+http-category-address = snapshot_rw,./relative/unix_path.sock
+http-category-address = trace_api,                               # disable
```

Comment:
Is this really supported and valid? When would it be useful? I guess something like
```ini
http-category-address = trace_api,127.0.0.1:8081
http-category-address = trace_api,[::1]:8081
http-category-address = trace_api,
```
But I'm not sure if that's the intention here or not.